### ISSUE TITLE
Guidance: Domain Events

### ISSUE THREAD

**Issue by mmmoli** [4lessandrodev/types-ddd#264]
I love this lib. Thank you for creating it 🙏 

**Is your feature request related to a problem? Please describe.**
'm stuck on handling domain events. Can you provide some guidance on how to set up subscriptions? Specifically subscribing to events across contexts.

**Describe the solution you'd like**
Any of the following:
- Documentation of subscriptions
- Updated example code
- written response

**Describe alternatives you've considered**
I've applied my knowledge of DDD, the documentation and your example project to build my own app. I can't quite work it out. `iHandle`, `IDomainEvent`, `DomainEvents`, `DomainHandler`, `Command`, bit lost.

**Additional context**
This guidance will really complement the fantastic work you've already done. You rock!

**Comment by mmmoli** [comment]
For reference, just spotted your finance-example-project:
https://github.com/4lessandrodev/finance-project-ddd/blob/main/src/modules/budget-box/domain/subscription/after-budget-box-deleted.subscription.ts

Huge help!

**Comment by 4lessandrodev** [comment]
Hey @mmmoli thanks for enjoying this Lib.
I believe there may be a version conflict. This Feature has been changed as can be checked in the [CHANGELOG file](https://github.com/4lessandrodev/rich-domain/blob/main/CHANGELOG.md). 
The project you mentioned uses version "types-ddd": "^2.12.0" there is a video explaining how to apply the events using the latest version of the lib, I believe it can be useful for you [Video Here](https://youtu.be/4w3v1hE-yqQ?t=652).

I hope I've helped
